src/tpm/tpm2_interface.c

```diff
@@ -74,8 +75,10 @@ int init_tpm2_connection(TSS2_SYS_CONTEXT ** sapi_ctx)
     // On failure, clean up initialization remnants to this point
     Tss2_Sys_Finalize(*sapi_ctx);
     free(*sapi_ctx);
-    Tss2_Tcti_Finalize(tcti_ctx);
-    free(tcti_ctx);
+    *sapi_ctx = NULL;
+    Tss2_TctiLdr_Finalize(&tcti_ctx);
+    if (tcti_ctx != NULL) free(tcti_ctx);
```

Comment:
Same issue here, I don't understand why we're setting ```tcti_ctx == NULL``` but not within the conditional on it being non-```NULL```.

Reply:
See the comment for the related issue above.